playbooks/roles/load-balancer-v2/tasks/main.yml

```diff
@@ -25,51 +38,112 @@
     - src: "haproxy.cfg.ctmpl"
       dest: "{{ consul_template_tmpl_dir }}/haproxy.cfg.ctmpl"
       mode: "0644"
+
+- name: Copy over Hashicorp Configuration Language files
+  # These HCL files instruct consul-watch where to find the templates,
+  # where to put them when they are filled in, and
+  # what to do after: reload HAProxy
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    mode: "{{ item.mode }}"
+  with_items:
     - src: "backend.map.hcl.j2"
       dest: "{{ consul_template_config_dir }}/backend.map.hcl"
       mode: "0644"
     - src: "haproxy.cfg.hcl.j2"
       dest: "{{ consul_template_config_dir }}/haproxy.cfg.hcl"
       mode: "0644"
+
+- name: Copy over Consul watcher scripts
+  # The template watcher watches for appserver configuration
+  # written by Ocim, and renders HAProxy configuration.
+  # The certifcate watcher watches for certificates written
+  # by the cert-manager, which get copied into HAProxy's cert dir.
+  # Both watchers reload HAProxy.
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    mode: "{{ item.mode }}"
+  with_items:
     - src: "consul-template-render.sh"
       dest: "/usr/local/sbin/consul-template-render.sh"
       mode: "0755"
     - src: "consul-certificates-render.sh"
       dest: "/usr/local/sbin/consul-certificates-render.sh"
       mode: "0755"
+
+- name: Copy over the local filesystem watcher
+  # The load balancer's own certificate is saved locally by certbot,
+  # so this watcher (not a Consul watcher!) moves it to
+  # HAProxy's cert dir as needed and reloads HAProxy.
+  template:
+    src: "cert-watcher"
+    dest: "/usr/local/sbin/cert-watcher"
+    mode: "0755"
+
+- name: Copy over systemd service files
+  # The watchers run as services.
+  template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    mode: "{{ item.mode }}"
+  with_items:
     - src: "consul-template-render.service"
       dest: "/lib/systemd/system/consul-template-render.service"
       mode: "0644"
     - src: "consul-certificates-render.service"
       dest: "/lib/systemd/system/consul-certificates-render.service"
       mode: "0644"
+    - src: "cert-watcher.service"
+      dest: "/lib/systemd/system/cert-watcher.service"
+      mode: "0644"
 
-# Haproxy refuses to start when there are no valid SSL certificates in the
-# /etc/haproxy directory, so we copy a bogus cert there.
-- name: Copy snakeoil certificate
+- name: Copy over snakeoil certificate
```

Comment:
Snakeoil certificates are present on Ubuntu under `/etc/ssl/certs/ssl-cert-snakeoil.pem` and `/etc/ssl/private/ssl-cert-snakeoil.key` if the `ssl-cert` package is present (I think this is a ubuntu default, so maybe we don't need to store it here). 
Do we need the same snakeoil certificate for all load balancers? Is is prossible to use the locally generated one?

Reply:
We don't need the same one; it should never be used for anything except convincing HAProxy that it is okay to start. I have fixed this.